src/main/java/org/opensearch/searchrelevance/transport/queryset/PostQuerySetTransportAction.java

```diff
@@ -79,10 +82,15 @@ protected void doExecute(Task task, PostQuerySetRequest request, ActionListener<
         }
 
         StepListener<Void> createIndexStep = new StepListener<>();
```

Comment:
you don't need this listener at all, and with this code putQuery will never be called actually.

Reply:
Done